src/main/java/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,100 @@
+import static racingcar.global.exception.ErrorMessage.DUPLICATE_NAME;
+import static racingcar.global.exception.ErrorMessage.INVALID_NAME_LIST_SIZE;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import racingcar.global.exception.RaceException;
+import racingcar.utils.Parser;
+
+public class Cars {
+    private List<Car> cars;
+
+    private Cars(final String userInput) {
+        List<Car> cars = Parser.parseStringToCarList(userInput);
+        Validator.validateCars(cars);
+        this.cars = cars;
+    }
+
+    public static Cars of(final String userInput) {
+        return new Cars(userInput);
+    }
+
+    public void tryMoveCars() {
+        for (Car car : cars) {
+            car.tryMove();
+        }
+    }
+
+    @Override
+    public String toString() {
+        List<String> carStates = generateCarStates();
+        return String.join(NEW_LINE.getSymbol(), carStates) + NEW_LINE.getSymbol();
+    }
+
+    public List<Car> findMaxPositionCars() {
+        Integer maxPosition = findMaxPosition();
+        return findSamePositionCars(maxPosition);
+    }
+
+    private List<String> generateCarStates() {
+        return cars.stream()
+                .map(Car::toString)
+                .toList();
+    }
+
+    private List<Car> findSamePositionCars(final Integer maxPosition) {
+        return cars.stream()
+                .filter(car -> car.isSamePosition(maxPosition))
+                .toList();
+    }
+
+    private Integer findMaxPosition() {
+        return findCarWithMaxPosition().getPosition();
+    }
+
+    private Car findCarWithMaxPosition() {
+        return cars.stream()
+                .max(Comparator.comparingInt(Car::getPosition))
+                .get();
+    }
+
+    private static class Validator {
+        private static void validateCars(final List<Car> cars) {
+            validateListSize(cars);
+            validateDuplicateNames(cars);
+        }
+
+        private static void validateListSize(final List<Car> cars) {
+            if (hasLessThanMinCars(cars)) {
+                throw RaceException.of(INVALID_NAME_LIST_SIZE);
+            }
+        }
+
+        private static boolean hasLessThanMinCars(final List<Car> cars) {
+            return cars.size() < MIN_CAR_SIZE.getValue();
+        }
+
+        private static void validateDuplicateNames(final List<Car> cars) {
+            if (hasDuplicateNames(cars)) {
+                throw RaceException.of(DUPLICATE_NAME);
+            }
+        }
+
+        private static boolean hasDuplicateNames(final List<Car> cars) {
+            Integer uniqueSize = countUniqueCarNames(cars);
+            return uniqueSize != cars.size();
+        }
+
+        private static Integer countUniqueCarNames(final List<Car> cars) {
+            return cars.stream()
+                    .map(Car::getName)
+                    .collect(Collectors.toSet())
+                    .size();
```

Comment:
이 부분도 너무 좋은 코드 같습니다!

Reply:
감사합니다 ! `getName()`을 불가피하게 사용하게 되는 코드여서 고민이 있었는데, 긍정적으로 평가해주셔서 다행이네요. 😊

Reply:
```suggestion
            return cars.stream()
                    .map(Car::getName)
                    .distinct()
                    .count();
```
Set을 만들지 않고 셀 수 있어요!
그런데 `int` 대신 `Integer`를 사용한 이유가 있으신가요?

Reply:
distinct내부적으로 set을 이용하는 걸로 알고있습니다.ㅎ 요부분도 취향의 영역이지 않을까 싶어요! ㅎ